inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/dataprovider/generic/DataProviderGenerator.java

```diff
@@ -0,0 +1,170 @@
+            cp.insertClassPath(new LoaderClassPath(loader));
+        }
+
+        CtClass provider = cp.get(GenericDataProviderTemplate.class.getName());
+        provider.setName(className);
+
+        for (String packageName : providerConfig.getImportedPackages()) {
+            cp.importPackage(packageName);
+        }
+
+        CtMethod method = provider.getDeclaredMethod("executeImpl");
+        method.setBody(buildProviderMethod(providerConfig));
+
+        return provider.toBytecode();
+    }
+
+    private String buildProviderMethod(ResolvedGenericDataProviderConfig providerConfig) {
+        StringBuilder methodBody = new StringBuilder("{");
+        if (providerConfig.getExpectedThisType() != null) {
+            buildVariableDefinition(methodBody, providerConfig.getExpectedThisType(), THIZ_VARIABLE, THIZ);
+        }
+        if (providerConfig.getExpectedReturnValueType() != null) {
+            buildVariableDefinition(methodBody, providerConfig.getExpectedReturnValueType(), RETURN_VALUE_VARIABLE, RETURN_VALUE);
+        }
+        if (providerConfig.getExpectedThrowableType() != null) {
+            String type = providerConfig.getExpectedThrowableType();
+            methodBody.append("if (!(").append(THROWN).append(" instanceof ").append(type).append(")){ return null; }\n");
+            buildVariableDefinition(methodBody, type, THROWN_VARIABLE, THROWN);
+        }
+        if (providerConfig.isUsesArgsArray()) {
+            buildVariableDefinition(methodBody, "Object[]", ARGS_VARIABLE, METHOD_ARGS);
+        }
+        providerConfig.getExpectedArgumentTypes().forEach((id, type) -> {
+            String value = METHOD_ARGS + "[" + id + "]";
+            String varName = ARG_VARIABLE_PREFIX + id;
+            buildVariableDefinition(methodBody, type, varName, value);
+        });
+        val additionalArgs = providerConfig.getAdditionalArgumentTypes();
+        for (int id = 0; id < additionalArgs.size(); id++) {
+            String value = ADDITIONAL_ARGS + "[" + id + "]";
+            val argsDef = additionalArgs.get(id);
+            val varName = argsDef.getLeft();
+            val varType = argsDef.getRight();
+            buildVariableDefinition(methodBody, varType, varName, value);
+        }
+        methodBody.append(providerConfig.getValueBody());
+
+        return methodBody.append("}").toString();
+    }
+
+    /**
+     * Builds a variable definition where another variable is casted from a Object variable.
+     * If the target type is a primitive, unboxing is performed.
+     *
+     * @param buf      the stringbuffer to write the assignment to
+     * @param type     the target type of the variable, can be a class or a primitive
+     * @param variable the name of the target variable
+     * @param value    the value to assign, can be a statement or another variable
+     */
+    private void buildVariableDefinition(StringBuilder buf, String type, String variable, String value) {
```

Comment:
same here, pls document what should be the output for the primitive and what for the object